src/main/java/ch/interlis/iox_j/wkb/RingCollector.java

```diff
@@ -0,0 +1,124 @@
+package ch.interlis.iox_j.wkb;
+
+import com.vividsolutions.jts.geom.Coordinate;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ */
+public class RingCollector {
+
+    private final boolean repairSelfTouchingRing;
+    private final LinkedList<List<LineSegment>> rings;
+    private final Map<Coordinate, Integer> coordinates2Segment;
+    private int currentRingIndex = -1;
```

Comment:
Q: Die Funktion von `currentRingIndex` habe ich noch nicht ganz verstanden. Wenn es immer der letzte Ring ist wäre die variable nicht nötig.

Reply:
Bei der Extraktion eines inneren Ringes, wird dieser innere Ring an das Ende der Liste hinzugefügt. Ein innerer Ring muss in der Liste immer nach dem äusseren Ring kommen. Im nach der Extraktion mit dem korrekten Ring weiter fahren zu können, merke ich mir hier dessen Index.